### PR TITLE
Add CI for monorepo template

### PR DESCRIPTION
**Pull request by dannymidnight**
This drops in a new monorepo template based on the following blog post. It's pretty basic and highlights that we me might need to add better support to our visualiser for this specific plugin (see this [issue](https://linear.app/buildkite/issue/GROW-762/add-first-class-support-for-the-monorepo-plugin) for more thoughts).

https://buildkite.com/blog/solving-ci-cd-in-monorepos-with-buildkite-s-official-plugin

Screenshot 2024-04-08 at 12 43 31 pm

### CODE REVIEW

File: monorepo-trigger/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+---
+title: CI for monorepo
+description: Conditionally trigger pipelines within a monorepo based on changes to file paths.
```
Comment (dannymidnight) [inline monorepo-trigger/README.md:3]: Not 100% sold on what I've written here. Very happy for suggestions and improvements.

File: monorepo-trigger/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
+title: CI for monorepo
```
Comment (dannymidnight) [inline monorepo-trigger/README.md:2]: I suspect there's room for a few monorepo templates so perhaps this should be slightly more specific.

File: monorepo-trigger/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+description: Conditionally trigger pipelines within a monorepo based on changes to file paths.
+author: Buildkite
+languages: []
+use_cases: ["Monorepo", "CI"]
+platforms: []
+tools: []
+primary_emojis: [":buildkite:"]
```
Comment (dannymidnight) [inline monorepo-trigger/README.md:9]: Open to alternatives for the main emoji.

Perhaps `:git:` + `':buildkite:` makes sense?

File: monorepo-trigger/pipeline.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+steps:
+  - label: ":git: Check changed files"
```
Comment (dannymidnight) [inline monorepo-trigger/pipeline.yaml:2]: Looking at this in the visualiser, I think something like `Conditionally trigger pipelines` might be more informative.

File: monorepo-trigger/README.md (hunk 1 above)
Comment (clbarrell) [reply monorepo-trigger/README.md:3]: for monorepo**s** ?

File: monorepo-trigger/README.md (hunk 3 above)
Comment (clbarrell) [reply monorepo-trigger/README.md:9]: +1 to git

just being buildkite probs isn't that isntructive

File: monorepo-trigger/pipeline.yaml (hunk 4 above)
Comment (clbarrell) [reply monorepo-trigger/pipeline.yaml:2]: It could also be referred to as doing a **diff**?

Comment (clbarrell) [reply monorepo-trigger/pipeline.yaml:2]: Or "Run against changed files"

File: monorepo-trigger/README.md (hunk 1 above)
Comment (dannymidnight) [reply monorepo-trigger/README.md:3]: I reckon plural might be more appropriate at the category level.

Comment (dannymidnight) [reply monorepo-trigger/README.md:3]: oh right, this was description. Yeah OK `for a monorepo` is better.

**Pull request merged**